environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -93,22 +137,65 @@
         }
     }
 
-    private ShutdownManager shutdownManager;
-    private Set<Metadata<Extension>> extensions;
+    private Map<String, WeldContainer> initializedContainers;
+
+    private String containerId;
 
-    private final String containerId;
+    private boolean discoveryEnabled = true;
+
+    private Set<String> beanClasses;
+
+    private List<Metadata<String>> selectedAlternatives;
+
+    private List<Metadata<String>> selectedAlternativeStereotypes;
+
+    private List<Metadata<String>> enabledInterceptors;
+
+    private List<Metadata<String>> enabledDecorators;
+
+    private Set<Metadata<Extension>> extensions;
 
     public Weld() {
         this(RegistrySingletonProvider.STATIC_INSTANCE);
     }
 
     /**
-     * Assign a unique ID to this Weld instance. This is required when running multiple Weld containers
-     * at once.
-     * @param containerId the unique ID for this container
+     *
+     * @param containerId The container identifier
+     * @see Weld#containerId(String)
      */
     public Weld(String containerId) {
         this.containerId = containerId;
+        this.initializedContainers = new HashMap<String, WeldContainer>();
+        this.beanClasses = new HashSet<String>();
+        this.selectedAlternatives = new ArrayList<Metadata<String>>();
+        this.selectedAlternativeStereotypes = new ArrayList<Metadata<String>>();
+        this.enabledInterceptors = new ArrayList<Metadata<String>>();
+        this.enabledDecorators = new ArrayList<Metadata<String>>();
+    }
+
+    /**
+     * Containers must have a unique identifier assigned when running multiple Weld instances at the same time.
+     *
+     * @param containerId
+     * @return self
+     */
+    public Weld containerId(String containerId) {
+        this.containerId = containerId;
+        return this;
+    }
+
+    /**
+     *
+     * @param classes The bean classes for the synthetic bean archive
+     * @return self
+     */
+    public Weld beanClasses(Class<?>... classes) {
+        this.beanClasses.clear();
+        for (Class<?> beanClass : classes) {
```

Comment:
We should probably consider BeanDeployArchive changes / evolvement to allow for Class<?> objects to be passed to Weld directly instead of doing Class<?> --> class name --> Class<?>